modal/functions.py

```diff
@@ -711,22 +714,25 @@ def call(self, *args, **kwargs) -> Awaitable[Any]:  # TODO: Generics/TypeVars
         else:
             return self._call_function(args, kwargs)
 
-    def __call__(self, *args, **kwargs) -> Any:  # TODO: Generics/TypeVars
-        if self._is_remote_cls_method:
-            return self.call(*args, **kwargs)
+    @synchronizer.nowrap
+    def __call__(wrapped_self, *args, **kwargs) -> Any:  # TODO: Generics/TypeVars
+        unwrapped_self = synchronizer._translate_in(wrapped_self)
```

Comment:
To access a bunch of instance-assigned private attributes within this method, since the self attribute is no longer "translated in" with the `nowrap` (`_is_remote_cls_method`, `_info`, `_self_obj`)

Reply:
I see. We could maybe make these public methods or getters to avoid this?